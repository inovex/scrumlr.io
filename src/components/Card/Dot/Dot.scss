@import '../../../constants/style';

$dot-color: $theme-color-primary;

.dot__small {
  width: 28px;
  height: 28px;
  line-height: 30px;
}

.dot__large {
  width: 32px;
  height: 32px;
  line-height: 32px;
}

.dot {
  display: block;
  text-align: center;
  border: 0;
  margin: 0;
  padding: 0;
  background: $dot-color;
  color: $theme-color-secondary;
  border-radius: 100%;
  position: relative;
  outline: none;
}

.dot__interactive {
  cursor: pointer;
}

.dot__folded-corner {
  width: 0;
  height: 0;
  position: absolute;
  top: 0;
  right: 0;
  border-width: 0;
  border-style: solid;
  background: hsla(260, 5%, 75%, 0.5);
  box-shadow: 2px -2px 0px #fff, 0 1px 1px rgba(0, 0, 0, 0.1),
    -1px 1px 1px rgba(0, 0, 0, 0.1);
  border-radius: 0 0 0 16px;
  border-color: $theme-color-background $theme-color-background transparent
    transparent;
  @include animate(border-width);
}

.dot.dot__small:hover .dot__folded-corner {
  border-width: 12px 0 0 12px;
}

.dot.dot__large:hover .dot__folded-corner {
  border-width: 18px 0 0 18px;
}
