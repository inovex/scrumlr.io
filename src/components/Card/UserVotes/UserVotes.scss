@import '../../../constants/style';

$icon__width: 24px;
$dot-color: #3991f4;

.user-votes {
  display: flex;
  align-items: center;
}

.user-votes__dots {
  margin: 0;
  padding: 0;
  list-style-type: none;
}

.user-votes__dots-item {
  display: inline-block;
}
