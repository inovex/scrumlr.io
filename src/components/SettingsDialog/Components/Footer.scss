@import "src/constants/style";
@import "../SettingsDialog";

// settings-dialog container
// $settings-dialog-container--sides-small: 32px;
// $settings-dialog-container--sides-large: 92px;
// $settings-dialog-container--bottom: 62px;

.container {
  margin-top: auto;
  padding: $padding--small $padding--medium;
  background-color: $color-lighter-gray;

  // overflow itsparent
  margin-left: -$settings-dialog-container--sides-small;
  width: calc(100% + 2 * $settings-dialog-container--sides-small);

  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
  align-items: center;

  // Text
  color: $menu-icon-background-color--dark;
  letter-spacing: $letter-spacing--medium;
  font-weight: 600;
}

.container button {
  padding: 0 $padding--medium;
  height: 36px;
  background-color: $menu-icon-background-color--dark;
  color: $color-white;
  border: none;
  border-radius: $rounded--full;
  font-weight: bold;
  letter-spacing: $letter-spacing--medium;
  cursor: pointer;

  &:hover {
    background-color: $hover-background-dark;
    color: $color-white;
  }
}

@media screen and (min-width: 920px) {
  .container {
    width: calc(100% + 2 * $settings-dialog-container--sides-large);
    margin-left: -$settings-dialog-container--sides-large;
  }
}

[theme="dark"] {
  .container {
    background-color: $color-dark-mode-note;
    // text
    color: $color-white;
  }
  .container button {
    background-color: $color-dark-mode;

    &:hover {
      background-color: $hover-background-dark;
    }
  }
}
