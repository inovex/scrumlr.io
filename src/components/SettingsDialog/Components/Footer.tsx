import {FC, PropsWithChildren} from "react";
import "./Footer.scss";

type FooterProps = {
  className?: string;
};

export const Footer: FC<PropsWithChildren<FooterProps>> = ({children, className}) => (
    <footer className={`container ${className || ""}`}>
      {children}
    </footer>
  );
