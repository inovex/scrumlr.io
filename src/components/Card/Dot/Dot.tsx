import * as classNames from 'classnames';
import * as React from 'react';

import './Dot.scss';

export type DotSize = 'small' | 'large';

export interface DotProps {
  size: DotSize;
  animate?: boolean;
  onClick?: (() => void) | undefined;

  [key: string]: any;
}

export class Dot extends React.Component<DotProps, {}> {
  static defaultProps: Partial<DotProps> = {
    animate: false,
    onClick: undefined
  };

  renderContent = () => {
    const { children, animate, onClick, size } = this.props;

    if (onClick) {
      return (
        <button
          type="button"
          className={classNames('dot', 'dot__interactive', `dot__${size}`)}
          onClick={onClick}
        >
          {animate && <span className="dot__folded-corner" />}

          {children}
        </button>
      );
    }

    return <div className={classNames('dot', `dot__${size}`)}>{children}</div>;
  };

  render() {
    const { className, size } = this.props;

    return (
      <span className={classNames(`dot__${size}`, className)}>
        {this.renderContent()}
      </span>
    );
  }
}

export default Dot;
