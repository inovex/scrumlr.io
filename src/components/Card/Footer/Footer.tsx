import * as React from 'react';

import UserVotes from '../UserVotes';
import Dot, { DotSize } from '../Dot';

export interface FooterProps {
  votable: boolean;
  votes: number | null;
  ownVotes: number;
  onDownvote: () => void;
  onUpvote: () => void;
  dotSize?: DotSize;
}

const defaultProps: Partial<FooterProps> = {
  dotSize: 'large'
};

export class Footer extends React.Component<FooterProps, {}> {
  static defaultProps: Partial<FooterProps> = defaultProps;

  render() {
    const {
      votable,
      ownVotes,
      onDownvote,
      onUpvote,
      votes,
      dotSize,
      children
    } = this.props;

    return (
      <footer className="card__footer">
        {children}
        <div className="card__footer-spacer" />
        <div>
          {votable && (
            <UserVotes
              votes={ownVotes}
              onDownvote={onDownvote}
              onUpvote={onUpvote}
            />
          )}

          {!votable && votes !== null && dotSize && (
            <Dot size={dotSize}>{votes}</Dot>
          )}
        </div>
      </footer>
    );
  }
}

export default Footer;
