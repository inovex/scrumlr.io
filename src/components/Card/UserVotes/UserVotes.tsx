import * as React from 'react';
import { range } from 'lodash';

import { AddDot, DownvoteDot } from '../Dot';
import './UserVotes.scss';

export interface UserVotesProps {
  onDownvote: () => void;
  onUpvote: () => void;
  votes?: number;
  maxDotsShown?: number;
}

const defaultProps: Partial<UserVotesProps> = {
  votes: 0,
  maxDotsShown: 5
};

export class UserVotes extends React.Component<UserVotesProps, {}> {
  static defaultProps: Partial<UserVotesProps> = defaultProps;

  renderDownvote(votes?: number) {
    return (
      <DownvoteDot onClick={() => this.props.onDownvote()}>{votes}</DownvoteDot>
    );
  }

  renderDots() {
    const { votes = 0, maxDotsShown } = this.props;

    if (maxDotsShown && votes && votes > maxDotsShown) {
      return this.renderDownvote(votes);
    }

    return (
      <ul className="user-votes__dots">
        {range(votes).map((x, i) => (
          <li key={votes - i} className="user-votes__dots-item">
            {this.renderDownvote()}
          </li>
        ))}
      </ul>
    );
  }

  render() {
    return (
      <span className="user-votes">
        {this.renderDots()}
        <AddDot onClick={() => this.props.onUpvote()} />
      </span>
    );
  }
}

export default UserVotes;
