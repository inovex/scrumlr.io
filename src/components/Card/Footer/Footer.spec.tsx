import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import { UserVotes } from '../UserVotes';
import { Dot, DotSize } from '../Dot';
import { Footer, FooterProps } from './Footer';

describe('<Footer />', () => {
  let wrapper: ShallowWrapper<FooterProps, {}>;
  let props: FooterProps;

  beforeEach(() => {
    props = {
      votable: false,
      ownVotes: 0,
      onDownvote: jest.fn(),
      onUpvote: jest.fn(),
      votes: 0
    };
  });

  it('should define default values', () => {
    expect(Footer.defaultProps).toMatchSnapshot();
  });

  it('should pass votes to UserVotes component', () => {
    const ownVotes = 20;
    wrapper = shallow(<Footer {...props} votable={true} ownVotes={ownVotes} />);
    expect(wrapper.find(UserVotes).prop('votes')).toEqual(ownVotes);
  });

  it('should not show votes if they are null', () => {
    wrapper = shallow(<Footer {...props} votes={null} />);
    expect(wrapper.find(Dot)).toHaveLength(0);
  });

  it('should show votes if they are not null', () => {
    const votes = 20;
    wrapper = shallow(<Footer {...props} votes={votes} />);
    expect(wrapper.find(Dot)).toHaveLength(1);
    expect(wrapper.find(Dot).prop('children')).toEqual(votes);
  });

  it('should pass onUpvote and onDownvote method to UserVotes', () => {
    wrapper = shallow(<Footer {...props} votable={true} />);
    const userVotes = wrapper.find(UserVotes);
    expect(userVotes.prop('onUpvote')).toBe(props.onUpvote);
    expect(userVotes.prop('onDownvote')).toBe(props.onDownvote);
  });

  it('should pass votes as children to dot component', () => {
    const votes = 10;
    wrapper = shallow(<Footer {...props} votes={votes} />);
    expect(wrapper.find(Dot).prop('children')).toEqual(votes);
  });

  it('should pass the dot size to dot component', () => {
    const dotSize: DotSize = 'small';
    wrapper = shallow(<Footer {...props} dotSize={dotSize} />);
    expect(wrapper.find(Dot).prop('size')).toEqual(dotSize);
  });
});
