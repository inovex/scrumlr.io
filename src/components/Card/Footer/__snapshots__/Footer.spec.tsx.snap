// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should define default values 1`] = `
Object {
  "dotSize": "large",
}
`;
